src/test/java/com/thebois/models/world/generation/noises/PerlinNoiseTests.java

```diff
@@ -0,0 +1,217 @@
+        final int seed1 = 0;
+        final int seed2 = 1;
+        perlinNoise.setSeed(seed1);
+        final float heightValue1;
+        final float heightValue2;
+
+        // Act
+        heightValue1 = perlinNoise.sample(x, y);
+        perlinNoise.setSeed(seed2);
+        heightValue2 = perlinNoise.sample(x, y);
+
+        // Assert
+        assertThat(heightValue1).isNotEqualTo(heightValue2);
+    }
+
+    @ParameterizedTest
+    @MethodSource("getPositions")
+    public void perlinNoiseDifferentResultWithSameValuesAndDifferentSettings(
+        final float x, final float y) {
+        // Arrange
+        final PerlinNoise perlinNoise1 = new PerlinNoise(1, 1, 1, 1);
+        final PerlinNoise perlinNoise2 = new PerlinNoise(2, 2, 2, 2);
+        final float heightValue1;
+        final float heightValue2;
+
+        // Act
+        heightValue1 = perlinNoise1.sample(x, y);
+        heightValue2 = perlinNoise2.sample(x, y);
+
+        // Assert
+        assertThat(heightValue1).isNotEqualTo(heightValue2);
+    }
+
+    @ParameterizedTest
+    @MethodSource("getPositions")
+    public void perlinNoiseReturnsZeroIfAmplitudeIs0(final float x, final float y) {
+        // Arrange
+        final PerlinNoise perlinNoise = new PerlinNoise(1, 0, 1, 1);
+        final float heightValue;
+        final float expectedHeightValue = 0;
+
+        // Act
+        heightValue = perlinNoise.sample(x, y);
+
+        // Assert
+        assertThat(heightValue).isEqualTo(expectedHeightValue);
+    }
+
+    @ParameterizedTest
+    @MethodSource("getPositions")
+    public void perlinNoiseReturnsZeroIfOctaveIs0(final float x, final float y) {
+        // Arrange
+        final PerlinNoise perlinNoise = new PerlinNoise(0, 1, 1, 1);
+        final float heightValue;
+        final float expectedHeightValue = 0;
+
+        // Act
+        heightValue = perlinNoise.sample(x, y);
+
+        // Assert
+        assertThat(heightValue).isEqualTo(expectedHeightValue);
+    }
+
+    @ParameterizedTest
+    @MethodSource("getPositions")
+    public void perlinNoiseReturnsNotZeroIfFrequencyIs0(final float x, final float y) {
+        // Arrange
+        final PerlinNoise perlinNoise = new PerlinNoise(1, 1, 0, 1);
+        final float heightValue;
+
+        // Act
+        heightValue = perlinNoise.sample(x, y);
+
+        // Assert
+        assertThat(heightValue).isNotEqualTo(0);
+    }
+
+    @ParameterizedTest
+    @MethodSource("getPositions")
+    public void perlinNoiseReturnsNotZeroIfPersistenceIs0(final float x, final float y) {
+        // Arrange
+        final PerlinNoise perlinNoise = new PerlinNoise(1, 1, 1, 0);
+        final float heightValue;
+
+        // Act
+        heightValue = perlinNoise.sample(x, y);
+
+        // Assert
+        assertThat(heightValue).isNotEqualTo(0);
+    }
+
+    @ParameterizedTest
+    @MethodSource("getPositions")
+    public void perlinNoiseReturnsSameResultIfPersistenceIs0AndOctavesIsDifferentAndRestSame(
```

Comment:
Really verbose name, which is great because I immediately understood what you were testing here, without even having to read the test. Nicely done!